state-chain/pallets/cf-pools/src/tests.rs

```diff
@@ -1437,3 +1377,185 @@ fn auto_sweeping() {
+			STABLE_ASSET,
+			Side::Sell,
+			1,
+			None,
+			0
+		));
+		assert_eq!(get_balance(&BOB), (10_000, 5000));
+	});
+}
+
+#[test]
+fn test_sweeping_when_updating_range_order() {
+	const ASSET: Asset = Asset::Flip;
+
+	let get_balance =
+		|lp| (MockBalance::get_balance(lp, ASSET), MockBalance::get_balance(lp, STABLE_ASSET));
+
+	new_test_ext().execute_with(|| {
+		// Turn off auto-sweeping
+		LimitOrderAutoSweepingThresholds::<Test>::mutate(|thresholds| {
+			thresholds.try_insert(ASSET, u128::MAX).unwrap();
+			thresholds.try_insert(STABLE_ASSET, u128::MAX).unwrap();
+		});
+
+		assert_ok!(LiquidityPools::new_pool(
+			RuntimeOrigin::root(),
+			ASSET,
+			STABLE_ASSET,
+			10_000, // 100bps fee
+			price_at_tick(0).unwrap(),
+		));
+
+		// Setup range orders
+		for (lp, amount) in [(ALICE, 5_000), (BOB, 5_000)] {
+			MockBalance::credit_account(&lp, ASSET, amount * 2);
+			MockBalance::credit_account(&lp, STABLE_ASSET, amount * 2);
+			assert_ok!(LiquidityPools::set_range_order(
+				RuntimeOrigin::signed(lp),
+				ASSET,
+				STABLE_ASSET,
+				1,
+				Some(-1..1),
+				RangeOrderSize::AssetAmounts {
+					minimum: PoolPairsMap { base: 0, quote: 0 },
+					maximum: PoolPairsMap { base: amount, quote: amount }
+				}
+			));
+			assert_ok!(LiquidityPools::set_range_order(
+				RuntimeOrigin::signed(lp),
+				ASSET,
+				STABLE_ASSET,
+				2,
+				Some(-1..1),
+				RangeOrderSize::AssetAmounts {
+					minimum: PoolPairsMap { base: 0, quote: 0 },
+					maximum: PoolPairsMap { base: amount, quote: amount }
+				}
+			));
+		}
+		assert_eq!(get_balance(&ALICE), (0, 0));
+		assert_eq!(get_balance(&BOB), (0, 0));
+
+		// Do a swap
+		assert!(LiquidityPools::swap_single_leg(STABLE_ASSET, ASSET, 10_000).is_ok());
+		assert!(LiquidityPools::swap_single_leg(ASSET, STABLE_ASSET, 10_000).is_ok());
+
+		// Confirm no sweeping has happened yet
+		assert_eq!(get_balance(&ALICE), (0, 0));
+		assert_eq!(get_balance(&BOB), (0, 0));
+
+		// Increase the range order should cause fee collection of both orders for that LP
+		MockBalance::credit_account(&ALICE, STABLE_ASSET, 5_000);
+		MockBalance::credit_account(&ALICE, ASSET, 5_000);
+		assert_ok!(LiquidityPools::set_range_order(
+			RuntimeOrigin::signed(ALICE),
+			ASSET,
+			STABLE_ASSET,
+			1,
+			None,
+			RangeOrderSize::AssetAmounts {
+				minimum: PoolPairsMap { base: 0, quote: 0 },
+				maximum: PoolPairsMap { base: 10_000, quote: 10_000 }
+			}
+		));
+		// NOTE: The collected fees would not be correct unless both orders got swept
+		const EXPECTED_FEES: u128 = 48;
+		assert_eq!(HistoricalEarnedFees::<Test>::get(ALICE, STABLE_ASSET), EXPECTED_FEES);
+		// But Bob's orders are not swept yet
+		assert_eq!(HistoricalEarnedFees::<Test>::get(BOB, STABLE_ASSET), 0);
+
+		// Decrease/cancel the range order should cause sweeping of it
+		assert_ok!(LiquidityPools::set_range_order(
+			RuntimeOrigin::signed(BOB),
+			ASSET,
+			STABLE_ASSET,
+			1,
+			None,
+			RangeOrderSize::Liquidity { liquidity: 0 }
```

Comment:
Just to make sure that we don't hit have a special case for 0 liquidity (which is essentially "cancelling" the order rather than "reducing" it), I think it would be better to use a number that's less than the linitial liquidity, but not 0?